src/node_sqlite.cc

```diff
@@ -506,6 +508,139 @@ void StatementSync::All(const FunctionCallbackInfo<Value>& args) {
+  if (r != SQLITE_ROW) {
+    CHECK_ERROR_OR_THROW(
+        env->isolate(), stmt->db_->Connection(), r, SQLITE_DONE, void());
+    sqlite3_reset(stmt->statement_);
+
+    LocalVector<Name> keys(isolate, {env->done_string(), env->value_string()});
+    LocalVector<Value> values(isolate,
+                              {Boolean::New(isolate, true), Null(isolate)});
+
+    Local<Object> result =
+        Object::New(isolate, Null(isolate), keys.data(), values.data(), 2);
+    args.GetReturnValue().Set(result);
+    return;
+  }
+
+  LocalVector<Name> row_keys(isolate);
+  row_keys.reserve(num_cols);
+  LocalVector<Value> row_values(isolate);
+  row_values.reserve(num_cols);
+  for (int i = 0; i < num_cols; ++i) {
+    Local<Name> key;
+    if (!stmt->ColumnNameToName(i).ToLocal(&key)) return;
+    Local<Value> val;
+    if (!stmt->ColumnToValue(i).ToLocal(&val)) return;
+    row_keys.emplace_back(key);
+    row_values.emplace_back(val);
+  }
+
+  Local<Object> row = Object::New(
+      isolate, Null(isolate), row_keys.data(), row_values.data(), num_cols);
+
+  LocalVector<Name> keys(isolate, {env->done_string(), env->value_string()});
+  LocalVector<Value> values(isolate, {Boolean::New(isolate, false), row});
+
+  Local<Object> result =
+      Object::New(isolate, Null(isolate), keys.data(), values.data(), 2);
+  args.GetReturnValue().Set(result);
+}
+
+void StatementSync::Iterate(const FunctionCallbackInfo<Value>& args) {
+  StatementSync* stmt;
+  ASSIGN_OR_RETURN_UNWRAP(&stmt, args.This());
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_ON_BAD_STATE(
+      env, stmt->IsFinalized(), "statement has been finalized");
+  auto isolate = env->isolate();
+  auto context = env->context();
+  int r = sqlite3_reset(stmt->statement_);
+  CHECK_ERROR_OR_THROW(
+      env->isolate(), stmt->db_->Connection(), r, SQLITE_OK, void());
+
+  if (!stmt->BindParams(args)) {
+    return;
+  }
+
+  IterateCaptureContext* capture_context = new IterateCaptureContext();
+  capture_context->num_cols = sqlite3_column_count(stmt->statement_);
+  capture_context->stmt = stmt;
+
+  Local<Function> next_func =
+      Function::New(context,
+                    StatementSync::IterateNextCallback,
+                    External::New(isolate, capture_context))
+          .ToLocalChecked();
+  Local<Function> return_func =
+      Function::New(context,
+                    StatementSync::IterateReturnCallback,
+                    External::New(isolate, capture_context))
+          .ToLocalChecked();
+
+  LocalVector<Name> keys(
+      isolate,
+      {env->next_string(), String::NewFromUtf8Literal(isolate, "return")});
+  LocalVector<Value> values(isolate, {next_func, return_func});
+
+  Local<Object> global = context->Global();
+  Local<Object> js_global_this =
+      global->Get(context, String::NewFromUtf8Literal(isolate, "globalThis"))
+          .ToLocalChecked()
+          .As<Object>();
+  Local<Object> js_iterator =
+      js_global_this
```

Comment:
Can't we directly get it from `global`?

Reply:
Is not it a better practice to get it from globalThis ?

Reply:
I mean from the `global` proxy. Your `global` variable here.

Reply:
`global.globalThis.Iterator` should be the same as `global.Iterator`

Reply:
Done